src/understand-tidb/transaction.md

```diff
@@ -1 +1,114 @@
 # Transaction
+
+The transaction engine in TiDB is responsible to provide `ACID` guarantee for all the read and write requests. It consists of the client/coordinator part in `tidb` repository and the server/participant part in `tikv` repository. This document is mainly about the `tidb` part.
+
+# The Architecture
+
+In `TiDB` the transaction write flow is like this:
+
+![transaction-architecture](../img/transaction-architecture.png)
+
+After the transaction is started in a session, all the reads and writes will use a snapshot to fetch data, and the written content will be buffered in the memory
+of the transaction. When the `commit` statement is received from the client, the `percolator` protocol will be used to commit these changes to the storage system.
+
+# The Interface
+
+In `tidb`, the [Transaction](https://github.com/pingcap/tidb/blob/af70762cd52519f025daa5e869ba37465a7fb311/kv/kv.go#L181) interface defines the commonly used transaction operations.
```

Comment:
```suggestion
In `TiDB`, the [`Transaction`](https://github.com/pingcap/tidb/blob/af70762cd52519f025daa5e869ba37465a7fb311/kv/kv.go#L181) interface defines the commonly used transaction operations.
```

Reply:
It's a problem, I often get confused about this. Sometimes I want to refer to the whole cluster components and `TiDB` is used, on the other hand, if the `tidb-server` is referred then `tidb` is used.

Reply:
🤔 Why the inline code format is used when it doesn't seem to be code?

Reply:
> It's a problem, I often get confused about this. Sometimes I want to refer to the whole cluster components and TiDB is used, on the other hand, if the tidb-server is referred then tidb is used.

@you06 @MyonKeminta 
What do you think about this

Reply:
idk.. May we ask for some help from our i18n team?
In my personal style, I usually write what I mean explicitly when there's no clear context, like "the TiDB cluster", "the TiDB node", "TiDB's code", etc..

Reply:
> thinking Why the inline code format is used when it doesn't seem to be code?

Because it's a name of an interface from code.